app/models/course/discussion/post.rb

```diff
@@ -98,4 +99,8 @@ def set_title
   def parent_topic_consistency
     errors.add(:topic_inconsistent) if parent && topic != parent.topic
   end
+
+  def reparent_children
+    children.update_all(parent_id: parent_id)
```

Comment:
@kxmbrian Can you check if this is in the same transaction of destroy ? If not, it's possible that the post destroy failed but the children go to the wrong parent.